src/array.ts

```diff
@@ -0,0 +1,34 @@
+import { Factory, FactoryConfig } from "./define";
+import { isFunction } from "./utils";
+
+type ArrayFactory<T> = (override?: FactoryConfig<T>) => T[];
+
+/**
+ * Define a new array factory function. The return value is a function that can be
+ * invoked as many times as needed to create an array of object of given type.
+ *
+ * @param factory An existing factory object.
+ * @param size Size of target array can either be a static value or a function that receives the
+ * invocation count as the only parameter.
+ */
+function array<Result>(
+  factory: Factory<Result>,
+  size: ((invocation: number) => number) | number = 5
```

Comment:
I like the idea of offering flexibility! I didn't see any examples of this usage in the tests or docs, would you mind minimally adding a test case for this to avoid any regressions?

Another option would be to only accept a number (`size: number = 5`) and you could achieve the same functionality with something like this?

```ts
const post = define<Post>(...);

const user = define<User>({
  name: "Me",
  posts: (i: number) => array(post, i)
})
```

However, in a test suite that calls a factory thousands of times (eg: `user()`), that means this would have to instantiate 1,000 `posts` on the 1,000th invocation of `user` which could end up being pretty expensive in a test suite. 

What are your thoughts? Do you think allowing `size` to be a function would be helpful?

Reply:
Thanks for pointing this out - after some thought I realised that this feature (accepting function as an argument) might be an overkill and it make it easy to do weird things - like someone writing `array(factory, random)` and generating 100k of items each time the test is being run :)

I think that your comment about only accepting a number as `array` size argument is the way to go - especially when it's rather easy to achieve the same result as I intended while being more explicit, which in turn will make it less likely to backfire.